main.go

```diff
@@ -15,6 +18,88 @@ import (
 var telemetryDB = make(map[string]TelemetryData)
 var listIPs = make(map[int]string)
 
+// Request is either:
+//   - Payload Ops (Ground)
+//   - Uplink/Downlink (Ground)
+func handleScenarioOne(c *gin.Context, r RedirectRequest) {
+	// Create a new HTTP request with the verb, URI and
+	// data as specified by the RedirectRequest
+	req, err := http.NewRequest(r.Verb, r.URI, bytes.NewReader([]byte(r.Data)))
+	if err != nil {
+		c.AbortWithStatus(http.StatusInternalServerError)
+		return
+	}
+
+	// Send this request using the default HTTP Client
+	// and receive a response
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		c.AbortWithStatus(http.StatusInternalServerError)
+		return
+	}
+
+	// Send the response back to the user
+	io.Copy(c.Writer, resp.Body)
+}
+
+// Request is either:
+//   - Payload Ops (Space)
+//   - CNDH (Space)
+//   - Uplink/Downlink (Space)
+//   - Payload Ops (Center)
+func handleScenarioTwo(c *gin.Context, r RedirectRequest) {
+
+}
+
+func receive(c *gin.Context) {
+	// Attempt to parse the incoming request's JSON into the "data" struct
+	var req RedirectRequest
+	if err := c.ShouldBindJSON(&req); err != nil {
+		// Abort internally stops Gin from contiuing to handle the request
+		c.AbortWithStatus(http.StatusBadRequest)
+		return
+	}
+
+	// Parse the IP from the RedirectRequest
+	//
+	// We do this by:
+	//   1. Calling url.ParseRequestURI which is a Golang
+	//      library function used to parse URLs
+	//   2. Call .Hostname() which returns the hostname of
+	//      the URL, this is where the IP is located
+	//
+	// E.g. "http://10.1.1.1:8080/telemetry/?id=5"
+	//   => "10.1.1.1"
+	ip := url.ParseRequestURI(req.URI).Hostname()
```

Comment:
URI.Hostname() returns a single string. This line is only getting 1 IP address (the destination IP address in the request)

On lines 86 you are referring to 'ip[0]' as if it is an array. Since 'ip' is just a string that will return the first char in the string. You need to check the entire thing.